Add section on efficiency considerations for zkLogin (#16917)
MystenLabs/sui: docs/content/concepts/cryptography/zklogin.mdx

Consequently, you should not store them persistently in an unsecure storage location, on any platform. For example, on browsers, use session storage instead of local storage to store the ephemeral key pair and the ZK proof. This is because session storage automatically clears its data when the browser session ends, while data in local storage persists indefinitely.

## Efficiency considerations

> Compared to traditional signatures, zkLogin signatures take a longer time to generate. For example, the prover that Mysten Labs maintains currently takes about three seconds to return a proof, which runs on a machine with 16 vCPUs and 64 GB RAM. Using more powerful machines, such as those with physical CPUs or graphics processing units (GPUs), can reduce the proving time further.

Reviewer: minor nit 
```suggestion
Compared to traditional signatures, zkLogin signatures take a longer time to generate. For example, the prover that Mysten Labs maintains typically takes about three seconds to return a proof, which runs on a machine with 16 vCPUs and 64 GB RAM. Using more powerful machines, such as those with physical CPUs or graphics processing units (GPUs), can reduce the proving time further.
```

> Carefully consider how many prover-requests your application needs. Broadly speaking, the right metric to consider is the number of active user sessions and not the number of signatures. This is because you can cache the same ZK proof and reuse it across the session, as previously explained. For example, if you expect a million active user sessions per day, then you need a prover that can handle one or two requests per second (RPS), assuming evenly distributed traffic.

Reviewer: ```suggestion
Carefully consider how many prover-requests your application needs to be able to handle. Broadly speaking, the right metric to consider is the number of active user sessions and not the number of signatures. This is because you can cache the same ZK proof and reuse it across the session, as previously explained. For example, if you expect a million active user sessions per day, then you need a prover that can handle one or two requests per second (RPS), assuming evenly distributed traffic.
```

> The Mysten Labs-maintained prover is set to auto-scale in order to handle traffic peaks. If you are not sure whether we can handle a specific number of requests, please reach out to us. Our plan is to horizontally scale the prover to handle any RPS you've in mind!

Reviewer 2: ```suggestion
The prover that Mysten Labs maintains is set to auto-scale to handle traffic peaks. If you are not sure whether Mysten Labs can handle a specific number of requests, please reach out to us. Our plan is to horizontally scale the prover to handle any RPS you have in mind.
```
Should include a mailto: link or similar for the "reach out to us" part. I'm not sure where that should go, but can find out if you don't either.

> The prover that Mysten Labs maintains is set to auto-scale to handle traffic peaks. If you are not sure whether Mysten Labs can handle a specific number of requests, please reach out to us on Discord. Our plan is to horizontally scale the prover to handle any RPS you have in mind.

Reviewer: ```suggestion
The prover that Mysten Labs maintains is set to auto-scale to handle traffic surges. If you are not sure whether Mysten Labs can handle a specific number of requests or expect a sudden spike in the number of prover requests your application needs to make, please reach out to us on Discord. Our plan is to horizontally scale the prover to handle any RPS you require.
```
